src/llv8.cc

```diff
@@ -1311,6 +1311,86 @@ void JSObject::Keys(std::vector<std::string>& keys, Error& err) {
   return;
 }
 
+
+void JSObject::Entries(std::vector<std::pair<Value, Value>>& entries,
+                       Error& err) {
+  entries.clear();
+
+  HeapObject map_obj = GetMap(err);
+
+  Map map(map_obj);
+
+  bool is_dict = map.IsDictionary(err);
+  if (err.Fail()) return;
+
+  if (is_dict) {
+    DictionaryEntries(entries, err);
+  } else {
+    DescriptorEntries(entries, map, err);
+  }
+
+  return;
+}
+
+void JSObject::DictionaryEntries(std::vector<std::pair<Value, Value>>& entries,
```

Comment:
You mean instead of being passed a vector by reference? (Not being passed something of type std::vector<>? I may have misunderstood this comment.)

I *think* it probably involves less copying as it is (C++ is good at catching me out though) and I added these functions to get round a performance issue so that was the focus when I put them in. It is a slight nuisance for the caller so possibly making the top level public function return a std::vector and the protected Dictionary and Descriptor functions use one they are given might hide the implementation better.